### ISSUE TITLE
Unable to install application using POST command

### ISSUE THREAD

**Issue by jpavlic** [facebook/WebDriverAgent#919]
Running Xcode 9.3, using simulator 11.3 iPhone 8, running the xcodebuild:

xcodebuild -project /Users/xxxxxx/Source/yyyyy/WebDriverAgent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination 'platform=iOS Simulator,id=62B1071F-0582-41EC-A81A-8D05C66A8DB1'  test
2018-04-30 17:35:19.173 xcodebuild[23221:207500]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:
/var/folders/sy/1y88f7dx1ybcvqn_f769k8k17r7s11/T/com.apple.dt.XCTest/IDETestRunSession-FC88B362-EF0C-4EA6-9B2F-42F23686825E/WebDriverAgentRunner-4B28CF80-F5AE-43E9-95FF-7965C0D40355/Session-WebDriverAgentRunner-2018-04-30_173519-gebWo6.log
2018-04-30 17:35:19.173 xcodebuild[23221:207477] [MT] IDETestOperationsObserverDebug: (E87EDC8B-D6B7-44E8-AAE0-851A9A4FF734) Beginning test session WebDriverAgentRunner-E87EDC8B-D6B7-44E8-AAE0-851A9A4FF734 at 2018-04-30 17:35:19.174 with Xcode 9E145 on target <DVTiPhoneSimulator: 0x7fcf0d255de0> {
		SimDevice: iPhone 8 (62B1071F-0582-41EC-A81A-8D05C66A8DB1, iOS 11.3, Booted)
} (11.3 (15E217))
=== BUILD TARGET WebDriverAgentLib OF PROJECT WebDriverAgent WITH CONFIGURATION Debug ===

Check dependencies

=== BUILD TARGET WebDriverAgentRunner OF PROJECT WebDriverAgent WITH CONFIGURATION Debug ===

Check dependencies

objc[23236]: Class VCWeakObjectHolder is implemented in both /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/PrivateFrameworks/AVConference.framework/Frameworks/ViceroyTrace.framework/ViceroyTrace (0x1293124d0) and /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/PrivateFrameworks/AVConference.framework/AVConference (0x126756e38). One of the two will be used. Which one is undefined.
2018-04-30 17:35:21.779337-0400 WebDriverAgentRunner-Runner[23236:207810] +[CATransaction synchronize] called within transaction
2018-04-30 17:35:21.816706-0400 WebDriverAgentRunner-Runner[23236:207810] Running in the background.
2018-04-30 17:35:21.819126-0400 WebDriverAgentRunner-Runner[23236:207810] Running tests...
Test Suite 'All tests' started at 2018-04-30 17:35:21.995
Test Suite 'WebDriverAgentLib.framework' started at 2018-04-30 17:35:21.996
Test Suite 'WebDriverAgentLib.framework' passed at 2018-04-30 17:35:21.996.
	 Executed 0 tests, with 0 failures (0 unexpected) in 0.000 (0.000) seconds
Test Suite 'WebDriverAgentRunner.xctest' started at 2018-04-30 17:35:21.997
Test Suite 'UITestingUITests' started at 2018-04-30 17:35:21.997
Test Case '-[UITestingUITests testRunner]' started.
    t =     0.00s Start Test at 2018-04-30 17:35:21.998
    t =     0.00s Set Up
2018-04-30 17:35:21.999860-0400 WebDriverAgentRunner-Runner[23236:207810] Built at Apr 30 2018 16:19:35
2018-04-30 17:35:22.015144-0400 WebDriverAgentRunner-Runner[23236:207810] ServerURLHere->http://xxx.xxx.xxx.xxx:8100<-ServerURLHere

And using the url with a POST command:

http://localhost:8100/session {"desiredCapabilities":{"bundleId":"com.xxxxxx.yyyyyyyyyyyy", "app":"/Users/xxxxx/Source/yyyyyy/build/zzzzzzzzz.app"}}

the results were:

    t =     9.98s Open com.xxxxxx.yyyyyyyyyyyy
    t =     9.98s     Launch com.xxxxxx.yyyyyyyyyyyy
2018-04-30 17:35:32.125867-0400 WebDriverAgentRunner-Runner[23236:207810] Enqueue Failure: Failed to launch com.xxxxxx.yyyyyyyyyyyy: The operation couldn’t be completed. Application "com.xxxxxx.yyyyyyyyyyyy" is unknown to FrontBoard. /Users/xxxxxx/Source/yyyyy/WebDriverAgent/WebDriverAgentRunner/UITestingUITests.m 37 1
2018-04-30 17:36:32.130050-0400 WebDriverAgentRunner-Runner[23236:207810] Enqueue Failure: Application 'com.xxxxxx.yyyyyyyyyyyy' does not have a process ID /Users/xxxxxx/Source/yyyyy/WebDriverAgent/WebDriverAgentRunner/UITestingUITests.m 37 1
    t =    70.13s         Wait for accessibility to load
2018-04-30 17:37:32.134589-0400 WebDriverAgentRunner-Runner[23236:207810] Enqueue Failure: App accessibility isn't loaded /Users/xxxxxx/Source/yyyyy/WebDriverAgent/WebDriverAgentRunner/UITestingUITests.m 37 1
2018-04-30 17:38:32.139951-0400 WebDriverAgentRunner-Runner[23236:207810] Enqueue Failure: Application '<XCUIApplicationImpl: 0x60800026a4c0 com.xxxxxx.yyyyyyyyyyyy at /Users/xxxxxx/Source/zzzzzz/build/yyyyyyyyyyyy.app>' failed to activate (current state: Not Running) /Users/xxxxxx/Source/yyyyy/WebDriverAgent/WebDriverAgentRunner/UITestingUITests.m 37 1

It will work fine if I drag the app in the simulator then run it, or install it using Xcode, but not through the POST command.

Any ideas?

**Comment by anushkmittal** [comment]
bump, same issue!

**Comment by marekcirkos** [comment]
Is the path and build arch correct?

**Comment by anushkmittal** [comment]
yes, I have been building on generic iOS device and trying it install it on the simulator.

**Comment by lanzelot1989** [comment]
Is there any new information concerning this, we face a similar problem when letting Appium install the .app.

**Comment by lanzelot1989** [comment]
Any progress on this?
